### PR TITLE
fix: use ipfs peerId as pin location

### PR DESCRIPTION
**Pull request by olizilla**
Cluster now provides the peerId for the underlying ipfs node in pin status response, so we fix things to use that as our pin location peer id. Previously we used the cluster peerId which should only be used for internal cluster admin. With this change users will be able to use the peerId in the /status response to connect to the ipfs node that has their stuff via 
`ipfs swarm connect <peerId>`

- Fix `toPin` to use `ipfsPeerId`
- Update tests to verify that status response for new content has a peerId of the ipfs node for one of the cluster nodes.
- Update ipfs-cluster client to [4.0.0](https://github.com/nftstorage/ipfs-cluster/releases/tag/v4.0.0)
- Update ipfs-cluster in docker-compose to [0.14.5-rc1](https://github.com/ipfs/ipfs-cluster/releases/tag/v0.14.5-rc1)
- Tweak test hooks to not wait for things to shut down before printing the test results overview, to reduce debugging friction.

Fixes: https://github.com/web3-storage/web3.storage/issues/414

TODO:
- [x] db migration script to update PinLocation peerIds
- [x] verify cluster infra is updated to ipfs-cluster >= 0.14.5-rc1
- [x] schedule read-only maintenance

License: (Apache-2.0 AND MIT)
Signed-off-by: Oli Evans <oli@tableflip.io>

### CODE REVIEW

**Comment by olizilla** [comment]
Looks like the `pin_location.peer_name` field is not being set correctly in staging and prod. It's currently being set to the peerId. I will fix that as part of this PR.

**Comment by olizilla** [comment]
The thrashing of the `peer_name` is coming from cluster; it returns either the peer name or the peerId for a node depending on the pin status. see: https://github.com/ipfs/ipfs-cluster/issues/1569

I think ideally that'd be something we fix on cluster. Either way, it's not really in scope for this PR, so I'll raise a seperate issue to follow it up.

**Comment by olizilla** [comment]
Tested locally. new pin locations created with ipfs peerId. Next steps are
- verify with hsanjaun what the plan is for rolling out cluster update... will we roll out the rc or wait till the next stable release

once that is done, test it out on staging:

- deploy to staging with api in read-only mode
- run sql migration
- enable write on staging api
- upload content to staging api
- verify the Pin Locations table has the expected number of cluster peers in and each one still has the ipfs peer id in the `peerId` column
- celebrate then do it again on prod.

**Review by vasco-santos — APPROVED** [review]
Awesome, this is looking good! 👍🏼 

We just need to figure out the remaining dependencies to move forward with release + maintenance window

**Comment by olizilla** [comment]
@hsanjuan do you have a rough estimate for when you'll release v0.14.5 of ipfs-cluster? No rush, it's just to know when we could schedule landing this PR.

**Comment by olizilla** [comment]
Verified, cluster is ready for us to roll this out

```sh
ipfs-cluster-ctl --host /dnsaddr/web3.storage.ipfscluster.io version
0.14.5
```

**Comment by vasco-santos** [comment]
@olizilla let's schedule a maintenance window for next week

**Comment by alanshaw** [comment]
[Scheduled for 2022-02-24 @ 14:30 UTC](https://status.web3.storage/incidents/sgfjvc50gm61)

File: packages/tools/docker/cluster/docker-compose.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ services:
         
     cluster0:
       container_name: cluster0
-      image: ipfs/ipfs-cluster:latest
+      image: ipfs/ipfs-cluster:v0.14.5-rc1
```
Comment (olizilla) [inline packages/tools/docker/cluster/docker-compose.yml:50]: @alanshaw I'm going to revert this change to the docker-compose to as the latest is now `v0.14.5` https://github.com/ipfs/ipfs-cluster/releases

File: packages/api/test/hooks.js [hunk 2]
```diff
@@ -53,18 +53,19 @@ export const mochaHooks = () => {
       await delay(2000)
     },
     async afterAll () {
+      // Note: not awaiting promises here so we see the test results overview sooner.
```
[the remark this answers is not in the document]
Comment (alanshaw) [reply packages/api/test/hooks.js:56]: I think no, we might have to revert this if we ever need that.

**Review by alanshaw — APPROVED** [review]
LGTM
